src/container/srv_target.c

```diff
@@ -1289,7 +1289,22 @@ cont_child_destroy_one(void *vin)
 	ABT_mutex_unlock(cont->sc_mutex);
 
 	/* nobody should see it again after eviction */
-	daos_lru_ref_evict_wait(tls->dt_cont_cache, &cont->sc_list);
+	/**
+	 * This function may yield, potentially creating a race condition with
+	 * rebuild operations. During rebuild migration, the container could be
+	 * reopened and restarted, which could result in EBUSY errors from
+	 * subsequent vos_cont_destroy() calls.
+	 *
+	 * To resolve this issue:
+	 * 1. We avoid container eviction during waiting periods
+	 * 2. Container lookup failures are guaranteed by checking the
+	 *    @sc_destroying flag before proceeding
+	 *
+	 * This design ensures consistency by preventing concurrent access
+	 * to containers marked for destruction.
+	 */
+	daos_lru_ref_noevict_wait(tls->dt_cont_cache, &cont->sc_list);
+	daos_lru_ref_evict(tls->dt_cont_cache, &cont->sc_list);
```

Comment:
Do we need to check "if (!llink->ll_evicted)" before calling `daos_lru_ref_evict()` since we may yield for wait?

Reply:
In theory this is possible，but currently there is no other caller which might evict container, and calling daos_lru_ref_evict() again here is not harmful now. will add extra check if PR need be refreshed for any reason.

Reply:
Add the check inside daos_lru_ref_evict()